software/motion.py

```diff
@@ -0,0 +1,40 @@
+import math
+import numpy as np
+import tkinter as tk
+import serial
+import struct
+
+class IRobotMotion:
+    def open(self):
+        pass
+    def close(self):
+        pass
+    def move(self, x_speed, y_speed, rot_speed):
+        pass
+
+class OmniMotionRobot(IRobotMotion):
+
+    def __init__(self):
+        self.serialObj = serial.Serial()
+
+    def open(self, serial_port):
```

Comment:
Why not have the serial port defined in the constructor. This class is not a factory for serial ports so it should stay constant for the class instance.